tools/decompile-json/src/main.cpp

```diff
@@ -1036,7 +1036,16 @@ int main(int argc, char *argv[]) {
       }
     } else if (fdecl) {
       if (auto func = lifter.DeclareEntity(*fdecl)) {
+        auto func_with_name = module.getFunction(name);
```

Comment:
I think the following heuristic could apply:

step 1:
iterate over all named things, and apply the following naming scheme: `_<zero-padded hex address to width 8>_name`. (use `std::setw(8) << std::setfill('0') << std::hex << addr` to the `std::stringstream`.

step 2:
iterate over all named things. If the named thing is an external declaration that is called, set its name to `<name>`.

step 3:
iterate over all named things. if the entity is not a declaration, and if the name isn't already taken, then give the thing the name `<name>`

step 4:
iterate over all named things. If the thing's name isn't `<name>` and if its name starts with `_`, and if nothing else is using the name, then give the name the name `<name>`.

Reply:
I updated the symbol naming heuristic along the same line and tested with a few binaries which should work. It goes through the functions and fixes the symbol after the first iteration.